### PR TITLE
MOTS-710: Retry the connection check on iOS

### PR DESCRIPTION
**Pull request by jkondrat**

### CODE REVIEW

File: mobile/ios/mobile.xcodeproj/xcshareddata/xcschemes/mobile.xcscheme [hunk 1]
```diff
@@ -99,6 +99,13 @@
             ReferencedContainer = "container:mobile.xcodeproj">
          </BuildableReference>
       </BuildableProductRunnable>
+      <EnvironmentVariables>
```
Comment (jkondrat) [inline mobile/ios/mobile.xcodeproj/xcshareddata/xcschemes/mobile.xcscheme]: Adding this to make logs easier to read with xcode

Comment (jkondrat) [reply mobile/ios/mobile.xcodeproj/xcshareddata/xcschemes/mobile.xcscheme]: https://github.com/facebook/react-native/issues/21030

File: mobile/src/utils/connection.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+      return connInfo;
+    });
+  }
+  return connInfo;
+}
+
+export function handleConnectivityChange(connInfo) {
+  if (isIos) {
+    // retry the connection check twice on ios
+    // https://github.com/facebook/react-native/issues/8615
```
Comment (jkondrat) [inline mobile/src/utils/connection.js]: https://github.com/facebook/react-native/issues/8615 this was most likely the cause

File: mobile/src/utils/connection.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+const isIos = Platform.OS === 'ios';
+
+export function setConnection(connInfo) {
+    dispatch(setConnectionState(connInfo));
+}
+
+export async function retryIfNotConnected(connInfo) {
+  if (connInfo.type === 'none' || connInfo.type === 'unknown') {
+    NetInfo.getConnectionInfo().then((connectionInfo) => {
+      return connInfo;
```
Comment (awalkowiak) [inline mobile/src/utils/connection.js]: You want to return connInfo here or connectionInfo?

**Pull request merged**